pennylane/ops/qubit.py

```diff
@@ -559,6 +694,15 @@ class BasisState(Operation):
     par_domain = "A"
     grad_method = None
 
+    @staticmethod
+    def decomposition(n, wires=None):
+        decomp_ops = []
+        for w, p in enumerate(n.flatten()):
+            if p == 1:
+                decomp_ops.append(PauliX(wires=wires[w]))
```

Comment:
Will `wires[w]` work if `wires` is one of the enums?

Reply:
`wires` will never be an `enum` here, it will always be a list of integers passed by the user. The enums are only used in `Operation.num_wires` as a specification.